leap_counter_extension/src/index.ts

```diff
@@ -0,0 +1,112 @@
+// This is one of three related extension examples that demonstrate
+// JupyterLab's provider-consumer pattern, where plugins can depend
+// on and reuse features from one another. The three packages that
+// make up the complete example are:
+//
+//   1. The step_counter package. This package holds a token, a
+//      class + an interface that make up a stock implementation of
+//      the "step_counter" service, and a provider plugin that
+//      makes an instance of the Counter available to JupyterLab
+//      as a service object.
+//   2. The step_counter_extension package, that holds a
+//      UI/interface in JupyterLab for users to count their steps that
+//      connects with/consumes the step_counter service object via a
+//      consumer plugin.
+//   3. (*) The leap_counter_extension package (this one), that holds an alternate
+//      way for users to count steps (a leap is 5 steps). Like the step_counter_extension
+//      package, this holds a UI/interface in JupyterLab, and a consumer
+//      plugin that also requests/consumes the step_counter service
+//      object. The leap_counter_extension package demonstrates how
+//      an unrelated plugin can depend on and reuse features from
+//      an existing plugin. Users can install either the
+//      step_counter_extension, the leap_counter_extension or both
+//      to get whichever features they prefer (with both reusing
+//      the step_counter service object).
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { Widget } from '@lumino/widgets';
+
+import { StepCounter} from "step_counter";
+
+// This widget holds the JupyterLab UI/interface that users will
+// see and interact with to count and view their steps.
+class LeapCounterWidget extends Widget {
+
+  leapButton: HTMLElement;
+  combinedStepCountLabel: HTMLElement;
+  counter: any;
```

Comment:
We should avoid `any` as an ani-pattern and  because it hides the important bit that this is `StepCounter`

```suggestion
  counter: StepCounter;
```

The three members should probably be private; it is of no importance in other examples but here distinctions between private/public might (?) be useful to highlight that these things are not to be shared with the outside ("You depend on LeapCounterWidget and want to touch the counter? Request it explicitly in requires, don't touch my counter!"

Reply:
This is done.